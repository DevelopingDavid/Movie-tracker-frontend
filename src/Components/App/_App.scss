* {
  margin: 0;
}

.App {
  text-align: center;
}

html { 
  background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url('./images/marvel-main.jpg') no-repeat center center fixed; 
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}

// body {
//   // background-size: center;
//   background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url('./images/marvel-main.jpg');
// }