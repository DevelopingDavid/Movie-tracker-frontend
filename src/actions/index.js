export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies,
});

export const loginUser = (loggedIn) => ({
  type: 'LOGIN_USER',
  loggedIn
});