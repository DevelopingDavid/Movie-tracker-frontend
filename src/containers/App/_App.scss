* {
  margin: 0;
}

html { 
  background: url('./images/marvel-main.jpg') no-repeat center center fixed;
  background-color: rgba(5, 5, 5, 0.815);
  background-blend-mode: multiply; 
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
}