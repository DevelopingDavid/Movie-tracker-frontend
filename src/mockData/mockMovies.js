export const mockMovies = [
  {
    "vote_count": 1128,
    "id": 166428,
    "video": false,
    "vote_average": 7.7,
    "title": "How to Train Your Dragon: The Hidden World",
    "popularity": 429.644,
    "poster_path": "/xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
    "original_language": "en",
    "original_title": "How to Train Your Dragon: The Hidden World",
    "genre_ids": [
      16,
      10751,
      12
    ],
    "backdrop_path": "/h3KN24PrOheHVYs9ypuOIdFBEpX.jpg",
    "adult": false,
    "overview": "As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothless’ discovery of an untamed, elusive mate draws the Night Fury away. When danger mounts at home and Hiccup’s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind.",
    "release_date": "2019-01-03"
  },
  {
    "vote_count": 2816,
    "id": 299537,
    "video": false,
    "vote_average": 7.3,
    "title": "Captain Marvel",
    "popularity": 373.277,
    "poster_path": "/AtsgWhDnHTq68L0lLsUrCnM7TjG.jpg",
    "original_language": "en",
    "original_title": "Captain Marvel",
    "genre_ids": [
      28,
      12,
      878
    ],
    "backdrop_path": "/w2PMyoyLU22YvrGK3smVM9fW1jj.jpg",
    "adult": false,
    "overview": "The story follows Carol Danvers as she becomes one of the universe’s most powerful heroes when Earth is caught in the middle of a galactic war between two alien races. Set in the 1990s, Captain Marvel is an all-new adventure from a previously unseen period in the history of the Marvel Cinematic Universe.",
    "release_date": "2019-03-06"
  },
  {
    "vote_count": 312,
    "id": 458723,
    "video": false,
    "vote_average": 7.4,
    "title": "Us",
    "popularity": 299.886,
    "poster_path": "/ux2dU1jQ2ACIMShzB3yP93Udpzc.jpg",
    "original_language": "en",
    "original_title": "Us",
    "genre_ids": [
      53,
      27
    ],
    "backdrop_path": "/ADJ6V8W96It4KElY2SPZvkKPBR.jpg",
    "adult": false,
    "overview": "Husband and wife Gabe and Adelaide Wilson take their kids to their beach house expecting to unplug and unwind with friends. But as night descends, their serenity turns to tension and chaos when some shocking visitors arrive uninvited.",
    "release_date": "2019-03-14"
  }
]