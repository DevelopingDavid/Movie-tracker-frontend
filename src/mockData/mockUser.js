export const mockUser = {
  id: 1,
  name: "david",
  password: "password",
  email: "david@gmail.com"
};