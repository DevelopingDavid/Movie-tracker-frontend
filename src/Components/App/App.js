import React, { Component } from 'react';
import { addMovies } from '../../actions';
import { connect } from 'react-redux';
import { url } from '../../apiURL';
import MovieContainer from '../../containers/MovieContainer/MovieContainer';
import Form from '../Form/Form';

export class App extends Component {

  componentDidMount() {
    this.fetchNowPlaying(url)
  }

  fetchNowPlaying = async (url) => {
    const response = await fetch(url);
    const data = await response.json();
    const movies = data.results;
    const cleanedMovies = this.cleanMovies(movies);
    this.props.addMovies(cleanedMovies);
  }

  cleanMovies = (movies) => {
    const newArray = movies.map(movie => {
      return {
        id: movie.id, 
        releaseDate: movie.release_date, 
        img: movie.poster_path, 
        title: movie.title, 
        rating: movie.vote_average
      }
    });
    return newArray;
  }

  render() {
    return (
      <div className="App">
        <h1>Movie Tracker</h1>
        <Form />
        <MovieContainer/>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies))
});

export default connect(null, mapDispatchToProps)(App);