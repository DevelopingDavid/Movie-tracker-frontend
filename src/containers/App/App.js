import React, { Component } from 'react';
import { addMovies, loginUser } from '../../actions';
import { connect } from 'react-redux';
import { url } from '../../apiURL';
import MovieContainer from '../MovieContainer/MovieContainer';
import { Route } from 'react-router-dom';
import SignIn from '../SignIn/SignIn';
import SignUp from '../SignUp/SignUp';
import Header from '../../containers/Header/Header';
import MoviePopup from '../MoviePopup/MoviePopup';
import Favorites from '../Favorites/Favorites';
import PropTypes from 'prop-types';

export const apiKey = ``;

export class App extends Component {

  componentDidMount() {
    this.fetchMarvelMovies(url);
    }

  fetchMarvelMovies = async (url) => {
    const response = await fetch(url);
    const data = await response.json();
    const movies = data.results;
    const cleanedMovies = this.cleanMovies(movies);
    this.props.addMovies(cleanedMovies);
  }

  cleanMovies = (movies) => {
    const newArray = movies.map(movie => {
      return {
        movie_id: movie.id, 
        title: movie.title, 
        poster_path: movie.poster_path, 
        release_date: movie.release_date, 
        vote_average: movie.vote_average,
        overview: movie.overview,
        backdrop_path: movie.backdrop_path
      }
    });
    return newArray;
  }

  render() {
    return (
      <div className="App">
        <Header />
        <Route exact path='/' component={SignIn} />
        <Route exact path='/sign-in' component={SignIn} />
        <Route exact path='/sign-up' component={SignUp} />
        <Route exact path='/movies' component={MovieContainer} />
        <Route exact path={`/${this.props.user.id}/favorites`} />
        <Route exact path='/movieInfo/:id' render={({match}) => {
          const { id } = match.params
          const foundMovie = this.props.movies.find((movie) => {
            return id == movie.movie_id
          });
          if (foundMovie) {
            return <MoviePopup foundMovie={foundMovie}/>
          }
        }}/> 
        <Route exact path='/:user_id/favorites' component={Favorites} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
  movies: state.movies
})

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies)),
  loginUser: (id, name) => dispatch(loginUser(id, name))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
  
App.propTypes = {
    addMovies: PropTypes.func.isRequired,
    loginUser: PropTypes.func.isRequired,
    movies: PropTypes.array.isRequired,
    user: PropTypes.object.isRequired
  }