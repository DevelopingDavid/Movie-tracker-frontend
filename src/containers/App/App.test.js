import React from 'react';
import { App, mapDispatchToProps, mapStateToProps } from './App';
import { shallow } from 'enzyme';
import { addMovies } from '../../actions';
import { mockMovies } from '../../mockData/mockMovies';
import { mockUser } from '../../mockData/mockUser';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( 
      <App user={mockUser}/>
    ) 
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using mapDispatchToProps', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addMovies([{
        id: 166428,
        release_date: "2019-01-03",
        poster_path: "/xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
        title: "How to Train Your Dragon: The Hidden World",
        vote_average: 7.8
      }]);

      const mappedProps = mapDispatchToProps(mockDispatch);
      
      mappedProps.addMovies([{
        id: 166428,
        release_date: "2019-01-03",
        poster_path: "/xvx4Yhf0DVH8G4LzNISpMfFBDy2.jpg",
        title: "How to Train Your Dragon: The Hidden World",
        vote_average: 7.8
      }]);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});