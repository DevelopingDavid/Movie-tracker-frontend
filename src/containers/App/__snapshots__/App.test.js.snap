// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    user={
      Object {
        "email": "david@gmail.com",
        "id": 1,
        "name": "david",
        "password": "password",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/sign-in"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/sign-up"
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/movies"
        />,
        <Route
          exact={true}
          path="/1/favorites"
        />,
        <Route
          exact={true}
          path="/movieInfo/:id"
          render={[Function]}
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/:user_id/favorites"
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/sign-in",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/sign-up",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/movies",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/1/favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/movieInfo/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/:user_id/favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/sign-in"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/sign-up"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/movies"
          />,
          <Route
            exact={true}
            path="/1/favorites"
          />,
          <Route
            exact={true}
            path="/movieInfo/:id"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/:user_id/favorites"
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/sign-in",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/sign-up",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/movies",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/1/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/movieInfo/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/:user_id/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
